package/tests/test_server.py

```diff
@@ -615,88 +615,145 @@ def test_format_pipeline_data_no_layers(pipeline, new_catalog_with_layers):
+                "node_5": {"id": "node_5", "layer": "model_input"},
+                "node_6": {"id": "node_6"},
+                "node_7": {"id": "node_7", "layer": "feature"},
+            },
+            {
+                "node_1": {"node_2"},
+                "node_2": {"node_3"},
+                "node_3": {"node_6"},
+                "node_4": {"node_5"},
+                "node_5": set(),
+                "node_6": {"node_7"},
+                "node_7": set(),
+            },
+            ["raw", "int", "feature", "model_input"],
+        ),
+        (
+            # fan-out-fan-in dependency
+            """
         node_1(layer=raw) -> node_2 -> node_3(layer=int) -> node_6 -> node_7(layer=feature)
                 |----------> node_4(layer=int) -> node_5(layer=model_input) --^
         """,
-        {"node_1": {"id": "node_1", "layer": "raw"}, "node_2": {"id": "node_2"},
-         "node_3": {"id": "node_3", "layer": "int"}, "node_4": {"id": "node_4", "layer": "int"},
-         "node_5": {"id": "node_5", "layer": "model_input"},
-         "node_6": {"id": "node_6"},
-         "node_7": {"id": "node_7", "layer": "feature"}},
-        {"node_1": {"node_2"}, "node_2": {"node_3"}, "node_3": {"node_6"},
-         "node_4": {"node_5"}, "node_5": {"node_7"}, "node_6": {"node_7"}, "node_7": set()},
-        ["raw", "int", "model_input", "feature"]
-    ),
-    (
-        # disjoint dependency: when two groups of layers have no direct dependencies,
-        # their order is determined by topological order first and alphabetical order second,
-        # which is the default of the toposort library. In the example below, toposort the layers
-        # will give [{c, d}, {b, a}], so it will be come [c, d, a, b] when flattened.
-        """
+            {
+                "node_1": {"id": "node_1", "layer": "raw"},
+                "node_2": {"id": "node_2"},
+                "node_3": {"id": "node_3", "layer": "int"},
+                "node_4": {"id": "node_4", "layer": "int"},
+                "node_5": {"id": "node_5", "layer": "model_input"},
+                "node_6": {"id": "node_6"},
+                "node_7": {"id": "node_7", "layer": "feature"},
+            },
+            {
+                "node_1": {"node_2"},
+                "node_2": {"node_3"},
+                "node_3": {"node_6"},
+                "node_4": {"node_5"},
+                "node_5": {"node_7"},
+                "node_6": {"node_7"},
+                "node_7": set(),
+            },
+            ["raw", "int", "model_input", "feature"],
+        ),
+        (
+            # disjoint dependency: when two groups of layers have no direct
+            # dependencies,their order is determined by topological order first and
+            # alphabetical order second, which is the default of the toposort library.
+            # In the example below, toposort the layers will give [{c, d}, {b, a}],
+            # so it will become [c, d, a, b] when flattened.
```

Comment:
broke these comments into one more line to respect line lengths <100 now that `black` indented them